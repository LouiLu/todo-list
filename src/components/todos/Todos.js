import React from 'react';
import TodoItem from './TodoItem';
import PropTypes from 'prop-types';

const Todos = ({ todos, markComplete, delTodo }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.length > 0 ? (
        todos.map(todo => (
          <TodoItem
            key={todo.id}
            todo={todo}
            markComplete={markComplete}
            delTodo={delTodo}
          />
        ))
      ) : (
        <div className="alert alert-info" role="alert">
          Add Some Todos to start with.
        </div>
      )}
    </ul>
  );
};

Todos.propTypes = {
  todos: PropTypes.array.isRequired,
  markComplete: PropTypes.func.isRequired,
  delTodo: PropTypes.func.isRequired
};

export default Todos;
