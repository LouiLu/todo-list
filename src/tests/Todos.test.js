import Todos from '../components/todos/Todos';

const testTodosArr = [
  {
    id: 1,
    title: 'delectus aut autem',
    completed: false
  },
  {
    id: 2,
    title: 'quis ut nam facilis et officia qui',
    completed: false
  },
  {
    id: 3,
    title: 'fugiat veniam minus',
    completed: false
  },
  {
    id: 4,
    title: 'et porro tempora',
    completed: true
  },
  {
    id: 5,
    title: 'laboriosam mollitia et enim quasi adipisci quia provident illum',
    completed: false
  }
];

it('renders correctly', () => {
  const spy = sinon.spy();
  const wrapper = shallow(<Todos todos={testTodosArr} delTodo={spy} />);

  expect(wrapper).toMatchSnapshot();
});
