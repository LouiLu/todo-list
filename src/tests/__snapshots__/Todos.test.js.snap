// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoItem
    delTodo={[Function]}
    key="1"
    todo={
      Object {
        "completed": false,
        "id": 1,
        "title": "delectus aut autem",
      }
    }
  />
  <TodoItem
    delTodo={[Function]}
    key="2"
    todo={
      Object {
        "completed": false,
        "id": 2,
        "title": "quis ut nam facilis et officia qui",
      }
    }
  />
  <TodoItem
    delTodo={[Function]}
    key="3"
    todo={
      Object {
        "completed": false,
        "id": 3,
        "title": "fugiat veniam minus",
      }
    }
  />
  <TodoItem
    delTodo={[Function]}
    key="4"
    todo={
      Object {
        "completed": true,
        "id": 4,
        "title": "et porro tempora",
      }
    }
  />
  <TodoItem
    delTodo={[Function]}
    key="5"
    todo={
      Object {
        "completed": false,
        "id": 5,
        "title": "laboriosam mollitia et enim quasi adipisci quia provident illum",
      }
    }
  />
</ul>
`;
